org.adempiere.base/src/org/compiere/acct/Doc_AllocationHdr.java

```diff
@@ -420,7 +420,8 @@ else if (line.getC_CashLine_ID() != 0)
 						getC_Currency_ID(), null, line.getAmtSource().negate());
 					if (fl != null && payment != null)
 						fl.setAD_Org_ID(payment.getAD_Org_ID());
-					allocPayAccounted = allocPayAccounted.add(fl.getAcctBalance().negate());
+					if (fl != null)
+						allocPayAccounted = allocPayAccounted.add(fl.getAcctBalance().negate());
```

Comment:
`fl` can't spontaneously become null after the first call, so what about this as slightly more efficient: 

```
if (fl != null) {
  if (payment != null) {
    fl.setAD_Org_ID(payment.getAD_Org_ID());
  }
  allocPayAccounted = allocPayAccounted.add(fl.getAcctBalance().negate());
}